Sources/Routing/Parameter/Parameter.swift

```diff
@@ -32,15 +33,28 @@ public protocol Parameter {
     /// - returns: An instance of the `ResolvedParameter` type if one could be created.
     /// - throws: Throws an error if a `ResolvedParameter` could not be created.
     static func resolveParameter(_ parameter: String) throws -> ResolvedParameter
+    
+    /// Creates a `PathComponent` for this type which can be used
+    /// when registering routes to a router.
+    static var parameter: PathComponent { get }
```

Comment:
Moved to be a requirement of the protocol so that if users conform their types to `Parameter`, they'll get a fixit indicating that they need to specify a `ParameterType`